GET2019-E19-1219/src/Assignment1/ShoppingCart.java

```diff
@@ -0,0 +1,222 @@
+package Assignment1;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Scanner;
+
+public class ShoppingCart {
+
+	/**
+	 * AllProductList store product available to purchase
+	 */
+	private ArrayList<Product> AllProductList = new ArrayList<Product>();
+
+	/**
+	 * Cart store product in user shopping cart
+	 */
+	private HashMap<String, Product> Cart = new HashMap<String, Product>();
+
+	public static void main(String[] args) {
+		// TODO Auto-generated method stub
+		ShoppingCart shcart = new ShoppingCart();
+
+		Scanner UserInput = new Scanner(System.in);
+		shcart.createProductList();
+
+		/**
+		 * To Control Program Execution
+		 */
+		while (true) {
+			Product product = new Product();
+			System.out.println("\n1.Add Product " + "\n2.Update Product" + "\n3.Remove Product" + "\n4.Show Cart"
+					+ "\n5.Genrate Bill" + "\n6.Exit");
+			System.out.println("Please Select Valid Option:- ");
+			int option = UserInput.nextInt();
+			if (option == 6) {
+				System.out.println("\nTHANK YOU");
+				break;
+			}
+			switch (option) {
+			case (1): {
+				shcart.showAllProduct();
+				System.out.print("Select Product Id :-");
+				int pid = UserInput.nextInt();
+				if (pid > 0 && pid <= 5) {
+					System.out.print("Enter Quantity :- ");
+					product.setProductQuantity(UserInput.nextInt());
+					if (product.getProductQuantity() > 0) {
+
+						product.setProductId(pid);
+						product.setProductName(shcart.AllProductList.get(pid - 1).ProductName);
+						product.setProductPrice(shcart.AllProductList.get(pid - 1).ProductPrice);
+						shcart.addProduct(product);
+					} else {
+						System.out.println("Invalid Quantity!!!!");
+					}
+				}
+				break;
+			}
+			case (2): {
+				shcart.showCart();
+				System.out.print("Select product id :-");
+				int pid = UserInput.nextInt();
+				if (pid > 0 && pid <= 5) {
+					System.out.print("Enter Quantity :- ");
+					product.setProductQuantity(UserInput.nextInt());
+					if (product.getProductQuantity() > 0) {
+						// product.ProductId = shcart.AllProductList.get(pid - 1).ProductId;
+						product.setProductId(pid);
+						shcart.updateProduct(product);
+					} else {
+						System.out.println("Invalid Quantity!!!!");
+					}
+				}
+
+				break;
+			}
+			case (3): {
+				shcart.showCart();
+				System.out.print("Select product id :-");
+				int pid = UserInput.nextInt();
+				if (pid > 0 && pid <= shcart.AllProductList.size()) {
+					product.setProductId(pid);
+					shcart.removeProduct(product);
+				}
+
+				break;
+			}
+			case (4): {
+				shcart.showCart();
+				break;
+			}
+			case (5): {
+				shcart.genrateBill();
+				break;
+			}
+			default: {
+				System.out.println("Invalid Option Selected!!!!");
+			}
+			}
+			;
+		}
+	}
+
+	/**
+	 * 
+	 * @param product
+	 *            this contain detail about product
+	 * 
+	 */
+
+	private void addProduct(Product product) {
+

```

Comment:
Use exception handling whenever it required.

Reply:
ok sir from now on i will use exception handling whenever required